### ISSUE TITLE
EF Core - Creating projection for the whole subrtree

### ISSUE THREAD

**Issue by marian-margeta** [ChilliCream/hotchocolate#2767]
**Describe the bug**
I am trying to update HC in my project from version 10 to 11 but I found several problems. Now I am not able to configure projections. I can illustrate this in a simple example bellow.

**To Reproduce**
I have the following 2 entities that are bound with simple 1:n relation 

```csharp
public class Folder
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public ICollection<File> Files { get; set; }
}

public class File
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Text { get; set; }
    public Folder Folder { get; set; }
}
```

I registered GraphQL services and types:

```csharp
public void ConfigureServices(IServiceCollection services)
{
  services.AddGraphQLServer()
    .AddQueryType<QueryType>()
    .AddType<FolderType>()
    .SetPagingOptions(new PagingOptions()
    {
        IncludeTotalCount = true
    })
    .AddProjections()
    .AddFiltering()
    .AddSorting();
}

public class FolderType : ObjectType<Folder>
{
    protected override void Configure(IObjectTypeDescriptor<Folder> descriptor)
    {
        descriptor.Field(x => x.Files)
            .UseOffsetPaging()
            .UseFiltering()
            .UseSorting();
    }
}

public class Query
{
    public IQueryable<Folder> GetFolders([ScopedService] MyDbContext myDbContext)
    {
        return myDbContext.Folder;
    }
}

public class QueryType : ObjectType<Query>
{
    protected override void Configure(IObjectTypeDescriptor<Query> descriptor)
    {
        descriptor.Field(x => x.GetFolders(default))
            .Type<FolderType>()
            .UseDbContext<MyDbContext>()
            .UseOffsetPaging()
            .UseProjection()
            .UseFiltering()
            .UseSorting();
    }
}
```

Now when I want to use this GraphQL query I get the error - `ICollection` cannot be created:  
request
```graphql
{
  folders {
    items {
      name
      files {
        items {
          name
        }
      }
    }
  }
}
```
response
```json
{
  "errors": [
    {
      "message": "Unexpected Execution Error",
      "locations": [
        {
          "line": 2,
          "column": 3
        }
      ],
      "path": [
        "folders"
      ],
      "extensions": {
        "message": "Type 'System.Collections.Generic.ICollection`1[GraphQLWeb.File]' does not have a default constructor (Parameter 'type')",
        "stackTrace": "   at System.Linq.Expressions.Expression.New(Type type)\r\n   at HotChocolate.Data.Projections.Expressions.ProjectionExpressionBuilder.CreateMemberInit(Type type, IEnumerable`1 expressions)\r\n   at HotChocolate.Data.Projections.Expressions.Handlers.QueryableProjectionFieldHandler.TryHandleLeave(QueryableProjectionContext context, ISelection selection, ISelectionVisitorAction& action)\r\n   at HotChocolate.Data.Projections.Expressions.Handlers.ProjectionFieldWrapper`1.TryHandleLeave(T context, ISelection selection, ISelectionVisitorAction& action)\r\n   at HotChocolate.Data.Projections.ProjectionVisitor`1.Leave(ISelection selection, TContext context)\r\n   at HotChocolate.Data.Projections.SelectionVisitor`1.Visit(ISelection selection, TContext context)\r\n   at HotChocolate.Data.Projections.ProjectionVisitor`1.Visit(ISelection selection, TContext context)\r\n   at HotChocolate.Data.Projections.SelectionVisitor`1.VisitChildren(IOutputField field, TContext context)\r\n   at HotChocolate.Data.Projections.SelectionVisitor`1.Visit(IOutputField field, TContext context)\r\n   at HotChocolate.Data.Projections.ProjectionVisitor`1.Visit(IOutputField field, TContext context)\r\n   at HotChocolate.Data.Projections.ProjectionVisitor`1.Visit(TContext context)\r\n   at HotChocolate.Data.Projections.Expressions.QueryableProjectionProvider.<CreateExecutor>g__ExecuteAsync|2_1[TEntityType](FieldDelegate next, IMiddlewareContext context)\r\n   at HotChocolate.Types.Pagination.PagingMiddleware.InvokeAsync(IMiddlewareContext context)\r\n   at HotChocolate.Utilities.MiddlewareCompiler`1.ExpressionHelper.AwaitTaskHelper(Task task)\r\n   at HotChocolate.Data.ToListMiddleware`1.InvokeAsync(IMiddlewareContext context)\r\n   at HotChocolate.Types.EntityFrameworkObjectFieldDescriptorExtensions.<>c__DisplayClass2_1`1.<<UseDbContext>b__4>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Types.EntityFrameworkObjectFieldDescriptorExtensions.<>c__DisplayClass2_1`1.<<UseDbContext>b__4>d.MoveNext()\r\n--- End of stack trace from previous location ---\r\n   at HotChocolate.Execution.Processing.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)\r\n   at HotChocolate.Execution.Processing.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)"
      }
    }
  ],
  "data": {
    "folders": null
  }
}
```

So I tried to changed it to `List<>`. Now it does not ends with error but it prints incorrect result (`files` field is empty but it shouldn't be). Also the generated SQL is not correct, it doesn't use projection at all:

request
```graphql
{
  folders {
    items {
      name
      files {
        items {
          name
        }
      }
    }
  }
}
```
response
```json
{
  "data": {
    "folders": {
      "items": [
        {
          "name": "xxx",
          "files": {
            "items": []
          }
        }
      ]
    }
  }
}
```
```sql
-- text was not requested 
SELECT "t"."Name", "t"."Id", "f0"."Id", "f0"."FolderId", "f0"."Name", "f0"."Text"
FROM (
    SELECT "f"."Name", "f"."Id"
    FROM "Folder" AS "f"
    LIMIT @__p_0
) AS "t"
LEFT JOIN "File" AS "f0" ON "t"."Id" = "f0"."FolderId"
ORDER BY "t"."Id", "f0"."Id"
```

According to the documentation for projections this should be supported:
```
The projection middleware will create a projection for the whole subtree of its field
```
but how to achive this?

**Desktop (please complete the following information):**
 - OS: Win 10
 - Version `11.0.5`

**Comment by PascalSenn** [comment]
@marian-margeta  Projections of connections is unfortunately not supported.

**Comment by marian-margeta** [comment]
@PascalSenn  Thank you for your reply. So I stay with version 10 for now. Is it going to be supported in 11?

**Comment by PascalSenn** [comment]
no it's not supported in 11 either. The problem is that we project the query into a queryable and paging needs to have a queryable by itself.
We need to figure out how to work around this, but at the moment it is a limitation

**Comment by marian-margeta** [comment]
I tried to remove paging `UseOffsetPaging()` in classes `QueryType` and `FolderType` and the projection of nested entity started to work. But I noticed a problem with filtering of nested entities. The SQL expression does not contain where condition on nested entity:
request
```graphql
{
  folders {
    id
    files (where: {id: {eq: 1}}) {
      id
    }
  }
}
```
response
```sql
SELECT "f0"."Id", "t"."Id"
FROM "Folder" AS "t"
LEFT JOIN "File" AS "f0" ON "t"."Id" = "f0"."FolderId"
ORDER BY "f0"."Id", "d"."Id"
```

The result is correct but `where` condition is apparently processed in memory. Is this a known limitation or bug?

**Comment by PascalSenn** [comment]
what versions are you using now?

**Comment by marian-margeta** [comment]
I tried it with version `11.0.0` and `11.0.6` using net5.0. In older versions `10.*` the filtering of nested entities works without problems.

**Comment by PascalSenn** [comment]
can you provide a minimal reproduction of this issue? We have this cases under tests, I wonder what went wrong here

**Comment by marian-margeta** [comment]
OK, I created repo with the code I am using [here](https://github.com/marian-margeta/GraphQLTest). All code related to Hot Chocolate is in [Startup.cs](https://github.com/marian-margeta/GraphQLTest/blob/master/Startup.cs)

**Comment by marian-margeta** [comment]
@PascalSenn  Do you think this is a bug or I incorrectly configured something?

**Comment by PascalSenn** [comment]
@marian-margeta  Sorry for letting you wait, just so much to do :) 
So it is indeed a bug. 
This pr should fix it: #2836
